### PR TITLE
Add renting

### PR DESCRIPTION
**Pull request by m-talbi**
- Added `Classroom`, `Book`, `Rental` classes.
- Added many-to-many relationship between `Books` and `Students` through `Rental` class.
- Added many-to-one relationship between `Students` and `classroom`.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,11 +1,18 @@
 require_relative 'person'
 
 class Student < Person
```
Comment (julie-ify) [inline student.rb:3]: - [ ] Kindly add the method for associating the student to a classroom. This method should serve as the `belongs-to` end of the relationship. See example.

  ```
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  attr_accessor :title, :author, :rentals
+
+  # @param [String] title
+  # @param [String] author
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (julie-ify) [inline book.rb]: - [ ] In this Book class kindly add the method for the has-many-through relationship with the person class using the Rental class.

  ```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
  ```

File: person.rb [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
   def can_use_services?
     of_age? && @parent_permission
   end
 
   private
 
+  # @return [Boolean]
   def of_age?
     @age >= 18
   end
```
Comment (julie-ify) [inline person.rb]: - [ ] In this Person class kindly add the method for the relationship with the book through the rental.

  ```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Kweeka1,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 4]
```diff
@@ -1,26 +1,33 @@
...
 
+  # @return [String] name
   def correct_name
     @name
   end
 
+  # @return [Boolean]
   def can_use_services?
     of_age? && @parent_permission
   end
```
Comment (gemmen29) [inline person.rb:33]: - [ ] Kindly add the method for adding rentals that represents the `many-to-many` relation between `Book` and `Person` through `Rental` inside the `Person` class as requested by the previous reviewer.
```ruby
def add_rental(book, date)
  Rental.new(date, book, self)
end
```

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Kweeka1, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Highlights💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Kweeka1,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**